joomla/joomla-cms
# Display a warning whenever an menu item's alias has changed

Display a warning message whenever a menu item's alias has been updated. The message suggests to the admin user to create a new redirection to the old alias in order to avoid 404 errors from existing urls.

Pull Request for Issue #13851.

### Summary of Changes
1. Added two methods to the redirect plugin: 
- onContentbeforeSave : Check if the menu item's alias has changed. If it's the case, then display a warning message that suggests to the admin user to create a redirection to the old alias.
- onContentPrepareForm : Alter menu item edit form whenever there's GET params (old_alias & new_alias) in order to set default values for the fields `old_url` & `new_url`
2. An enable/disable field (`Watch Alias`) has been added to the plugin config form if there's need a disable this feature.

### Testing Instructions
First, make sure that the redirect plugin is enabled.

When enabled :
1. Edit an existing menu item and change the alias. Then hit the Save button.
2. A warning message should be displayed that says : "You should create a new redirection to avoid 404 errors that could be generated by the old alias's urls : **Create a new redirection**"
3. Click on the link **Create a new redirection** and you should be redirected to a new redirection form with the two fields `old_url` & `new_url` filled with default values. Hit the Save button.
4. Access the old URL in frontend (that uses the old alias) and you should be redirected to the new one.

When disabled :
1. Go the redirect plugin config form and disable `Watch Alias`
2. Back to the menu manager,  Edit an existing menu item and change the alias. Then hit the Save button.
3. There will be no warning message.

### Documentation Changes Required
No.
